test/e2e/route_advertisements.go

```diff
@@ -516,6 +524,10 @@ var _ = ginkgo.DescribeTableSubtree("BGP: isolation between advertised networks"
 		var ra *rav1.RouteAdvertisements
 		var hostNetworkPort int
 		ginkgo.BeforeEach(func() {
+			// Adjust IP family constants based on cluster support
+			if !isIPv4Supported(f.ClientSet) && isIPv6Supported(f.ClientSet) {
+				ipFamilyV6 = ipFamilyV4
```

Comment:
Can we do something a bit less confusing please? :D 
Maybe it is time to stop relying on the ipFamilyIndex and just use `utilnet.[IPv4|IPv6]`?